text/0124-documents-api.md

```diff
@@ -20,21 +20,25 @@ Documents are stored inside indexes.
 
 Manipulate documents of a Meilisearch index.
 
-- [3.1.1. `GET` - `indexes/:index_uid/documents`](#311-get---indexesindexuiddocuments)
+- [3.1.1. `GET` - `indexes/:index_uid/documents`](#311-get-indexesindexuiddocuments-and-post-indexesindexuiddocumentsfetch)
 - [3.1.2. `GET` - `indexes/:index_uid/documents/:document_id`](#312-get---indexesindexuiddocumentsdocumentid)
 - [3.1.3. `POST` - `indexes/:index_uid/documents`](#313-post---indexesindexuiddocuments)
 - [3.1.4. `PUT` - `indexes/:index_uid/documents`](#314-put---indexesindexuiddocuments)
 - [3.1.5. `DELETE` - `indexes/:index_uid/documents`](#315-delete---indexesindexuiddocuments)
 - [3.1.6. `DELETE` - `indexes/:index_uid/documents/:document_id`](#316-delete---indexesindexuiddocumentsdocumentid)
 - [3.1.7. `POST` - `indexes/:index_uid/documents/delete-batch`](#317-post---indexesindexuiddocumentsdelete-batch)
 
-#### 3.1.1. `GET` - `indexes/:index_uid/documents`
+#### 3.1.1. `GET` - `indexes/:index_uid/documents` and `POST` - `indexes/:index_uid/documents/fetch`
```

Comment:
An unsolicited comment/feedback/question: is it wise to have two endpoints that do almost exactly the same thing, but in slightly different ways? 

From a documentation perspective, it makes it harder for us to provide clear guidance on recommended practices and best use cases: "you can use A to do X, but you can also use B. B is slightly better for Y, but A works fine as well." This often results in users having to spend more time than they would want to on making a decision that might not be that important.

Reply:
Yeah, personally, I think we should deprecate the `GET` version of the route, same for the search.
To me, basically;
- The get route can be slightly easier to use with `curl`
- The post route is better on absolutely everything else (like sending big filters or using the array syntax)

Reply:
Deprecating `GET` would be fine for us. In case this is not possible (e.g. API stability and versioning), would be to get clear confirmation from product and/or engine teams that we can recommend users to prefer `POST` and discourage `GET` usage.

IMHO, with this change `GET` becomes a net negative. Being slightly easier to use in a relatively small use-case is outweighed by the effort required in choosing between similar options.

Reply:
@guimachiavelli totally fine on our side to recommend users to use `POST` instead of `GET`. We can even go so far as to warn users that `GET` will be deprecated in the near future.

Reply:
Brilliant, @macraig, thanks! We'll keep this in mind when documenting these changes and add you as a one of the reviewers to ensure our recommendations are aligned with the API design's intent 👍

Reply:
@macraig I think we could deprecate it for v1.3, given the feedback we receive until then. I like the current solution proposed here, indicating users to favor the new one and that we will deprecate it in the future.

Deprecating the GET search endpoint should be treated as another project, we first need to conduct some technical research to verify if the post one can be safely used for caching mechanism

Reply:
Why doesn't the same concern apply to the get documents route?

Also, since we had no analytics on the usage of the get route, are we not afraid of deprecating it _that early_ without having any idea of its usage?

(I'm just throwing some ideas; personally, I would like to deprecate both routes asap so users have the time to update their code until the v2 :joy:)

Reply:
There is no urgency to depreciate it indeed, it is far from being a priority today, we can give ourselves time. What matters now, as pointed out by @guimachiavelli, is to eliminate the doubt among users between using GET or POST to fetch documents and thus simplify their use of the documentation and ultimately the product.

Reply:
For what it's worth, we'd also be very happy if `GET` `/search` were to be deprecated.

Reply:
So, just to be sure are we going to deprecate the `GET index/{index_uid}/documents` right away or are we waiting until the v1.3?

Reply:
@brunoocasali We won't mark it as deprecated for v1.2, we are waiting for v1.3

Reply:
Should we (integrations) follow that? Cause I don't see a world where waiting til v1.3 will make much difference for the users, a deprecation notice is not the same as "this method does not work anymore. Take this exception in your face instead".

Reply:
Once we deprecate it, it'll continue to work until the v2 of meilisearch, right?
The user will never get an exception, I think? (but maybe you could deprecate it in the integration and get rid of the method _before_ meilisearch v2 :thinking:)